src/daemon/collector/overload/post-sponsored-data.js

```diff
@@ -0,0 +1,38 @@
+import { v4 as uuidv4 } from "uuid";
+import { fetchWaistForSponsoredItem } from "./waist-requests";
+
+/**
+ * @typedef {Object} waistVariablesForSponsoredItem
+ * @property {String} fb_dtsg
+ * @property {String} variables
+ * @property {String} doc_id
+ * @property {String} adId
+ * @property {{
+ * fb_dtsg: String,
+ * variables: String,
+ * doc_id: String,
+ * adId: String,
+ * fields: String{ ad_id: String, client_token: String, request_id: String }
+ * }} fields
+ *
+ * @typedef {Object} advertData
+ */
+
+/**
+ * Sends request to generate waist data for the sponsored post
+ * and then uses waist response to send both sponsored data and waist data
+ * to our DB
+ * @param {Object} waistVariablesForSponsoredItem variables required to get waist data
+ * @param {Object} advertData full advertised data
+ */
+export const postSponsoredData = (waistVariablesForSponsoredItem, advertData) => {
+  fetchWaistForSponsoredItem(waistVariablesForSponsoredItem).then((waistData) => {
```

Comment:
Should this return? Or await with a try/catch? I'd be worried about unhandled rejections here.

Reply:
I added a `catch` clause only to this one because so far we don't really have a way of handling these failed post requests.